src/integration-test/groovy/org/pih/warehouse/api/spec/TestingDatesApiSpec.groovy

```diff
@@ -0,0 +1,113 @@
+        "2000-01-01T000000000Z"     || "2000-01-01T00:00:00Z" | "Time with no separator"
+        "2000-01-01T00000000Z"      || "2000-01-01T00:00:00Z" | "Time with no separator, shorter millis"
+        "2000-01-01T0000000Z"       || "2000-01-01T00:00:00Z" | "Time with no separator, shorter millis"
+        "2000-01-01T000000Z"        || "2000-01-01T00:00:00Z" | "Time with no separator, no millis"
+        "2000-01-01T0000Z"          || "2000-01-01T00:00:00Z" | "Time with no separator, no seconds"
+        "2000-01-01T00:00:00:000Z"  || "2000-01-01T00:00:00Z" | "Time with millis replace '.' with ':'"
+        "2000-01-01T00:00:00:00Z"   || "2000-01-01T00:00:00Z" | "Time with millis replace '.' with ':', shorter millis"
+        "2000-01-01T00:00:00:0Z"    || "2000-01-01T00:00:00Z" | "Time with millis replace '.' with ':', shorter millis"
+        // It can handle various timezone inputs. Yay!
+        "2000-01-01T00:00:00-05"    || "2000-01-01T05:00:00Z" | "Timezone conforming to X format"
+        "2000-01-01T00:00:00-0500"  || "2000-01-01T05:00:00Z" | "Timezone conforming to XX format"
+        "2000-01-01T00:00:00-05:00" || "2000-01-01T05:00:00Z" | "Timezone conforming to XXX format"
+    }
+
+    void 'test LocalDate: #givenDate'() {
+        given:
+        JSONObject request = new JSONObject().put('myLocalDate', givenDate)
+
+        when:
+        JsonPath response = api.testDatesOK(request)
+
+        then:
+        assert response.get("data.myLocalDate") == expectedResult
+
+        where:
+        givenDate    || expectedResult | scenario
+        "2000-01-01" || "2000-01-01"   | "Date with dashes"
+        "2000 01 01" || "2000-01-01"   | "Date with spaces"
+        "2000/01/01" || "2000-01-01"   | "Date with slashes"
+        "20000101"   || "2000-01-01"   | "Date with no separator"
+    }
+
+    void 'test Date: #scenario'() {
+        given:
+        JSONObject request = new JSONObject().put('myDate', givenDate)
+
+        when:
+        JsonPath response = api.testDatesOK(request)
+
+        then:
+        assert response.get("data.myDate") == expectedResult
+        assert response.get("data.myDateToString") == expectedToStringResult
+
+        // See how the resulting format changes depending on whether you call Date.toString() or not in the domain's
+        // toJson() method. This inconsistency is confusing and error-prone! Not a huge issue though because we'd
+        // notice immediately if we accidentally did Date.toString() because it'd break the frontend.
+        where:
+        givenDate                    || expectedResult         | expectedToStringResult         | scenario
+        // See how we input no tz information, but the server decides to treat it as local time (which for me currently
+        // is CST). If someone runs this test against a server in a different tz, it'll fail. Bad!
+        // To fix this, we need to add "TimeZone.setDefault(TimeZone.getTimeZone(ZoneOffset.UTC))" to BootStrap.groovy
```

Comment:
This scenario is demonstrable right here in this PR. The GitHub Action fails because it's running in a different timezone (UTC) https://github.com/openboxes/openboxes/actions/runs/13077910108/job/36494469201?pr=5022